lib/cylc/network/https/base_client.py

```diff
@@ -64,28 +64,55 @@ def __init__(self, suite, owner=USER, host=None, port=None, timeout=None,
         self.server_cert = None
         self.auth = None
 
-    def call_server_func(self, category, fname, **fargs):
-        """Call server_object.fname(*fargs, **fargs)."""
+    def _compile_url(self, category, func_dict, host):
+        payload = func_dict.pop("payload", None)
+        method = func_dict.pop("method", self.METHOD)
+        function = func_dict.pop("function", None)
+        url = 'https://%s:%s/%s/%s' % (host, self.port, category, function)
+        # If there are any parameters left in the dict after popping,
+        # append them to the url.
+        if func_dict:
+            import urllib
+            params = urllib.urlencode(func_dict, doseq=True)
+            url += "?" + params
+        request = {"url": url, "payload": payload, "method": method}
+        return request
+
+    def call_server_func(self, category, *func_dicts, **fargs):
+        """func_dict is a dictionary of command names (fnames)
+        and arguments to that command"""
+        # Deal with the case of one func_dict/function name passed
+        # by converting them to the generic case: a dictionary of
+        # a single function and its function arguments.
+        if isinstance(func_dicts[0], str):
+            function = func_dicts[0]
+            func_dict = {"function": function}
+            func_dict.update(fargs)
+
         if self.host is None and self.port is not None:
             self.host = get_hostname()
         try:
             self._load_contact_info()
         except (IOError, ValueError, SuiteServiceFileError):
             raise ConnectionInfoError(self.suite)
         handle_proxies()
-        payload = fargs.pop("payload", None)
-        method = fargs.pop("method", self.METHOD)
         host = self.host
         if host == "localhost":
             host = get_hostname().split(".")[0]
-        url = 'https://%s:%s/%s/%s' % (host, self.port, category, fname)
-        if fargs:
-            import urllib
-            params = urllib.urlencode(fargs, doseq=True)
-            url += "?" + params
-        return self._get_data_from_url(url, payload, method=method)
 
-    def _get_data_from_url(self, url, json_data, method=None):
+        http_requests = []
+        try:
+            # dictionary containing: url, payload, method
+            request = self._compile_url(category, func_dict, host)
+            http_requests.append(request)
```

Comment:
It is perhaps unfortunate that the module providing the main function is called `requests`. I suppose we should use variable names that do not easily confuse with the name of that module?

Suggest something like `http_request_items` (for plural) and `http_request_item` for singular.